docs/Customization/Custom-Core.rst

```diff
@@ -0,0 +1,267 @@
+.. _custom_core:
+
+Adding a custom core
+====================
+
+You may want to integrate a custom RISC-V core into the Chipyard framework. This documentation page provides a step-to-step
+instruction on how to achieve this.
+
+.. note:: 
+
+    RoCC is currently not supported by cores other than Rocket and BOOM. Please use Rocket or BOOM as the RoCC base core if you need to use RoCC.
+
+
+Wrap Verilog Module with Blackbox (Optional)
+--------------------------------------------
+
+Since Chipyard uses Scala and Chisel, if the top module of your core is not in Chisel, you will first need to create a Verilog
+blackbox for it so that it can be processed by Chipyard. See :ref:`incorporating-verilog-blocks` for instructions.
+
+Create Parameter Case Classes
+-----------------------------
+
+Chipyard will generate a core for every ``InstantiableTileParams`` object it discovered in the ``TilesLocated(InSubsystem)`` key.
+This object is derived from``TileParams``, a trait containing the information needed to create a tile. All cores must have
+their own implementation of ``InstantiableTileParams``, as well as ``CoreParams`` which is passed as a field in ``TileParams``.
+
+``TileParams`` holds the parameters for the tile, which are the same for every generated core, while ``CoreParams`` 
+contains the parameters for individual cores. They must be implemented as case classes with fields that can be overridden by 
```

Comment:
This could use some clarifications (or maybe examples)

Reply:
I misunderstood ``TileParams`` and ``CoreParams``. I changed the wording and it should be good now.